cmd/zc_attr_filter_windows.go

```diff
@@ -0,0 +1,105 @@
+// Copyright © 2017 Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"path/filepath"
+	"syscall"
+	"strings"
+)
+
+type attrFilter struct {
+	fileAttributes uint32
+	filePath       string
+	resultIfMatch  bool
+}
+
+func (f *attrFilter) doesSupportThisOS() (msg string, supported bool) {
+	msg = ""
+	supported = true
+	return
+}
+
+// TODO: review handling of filename for long path support
+func (f *attrFilter) doesPass(storedObject storedObject) bool {
+	fileName := filepath.Join(f.filePath, storedObject.name)
+	lpFileName, err := syscall.UTF16PtrFromString(fileName)
+
+	// If it fails to retrive the pointer from file path, let the filter pass
+	if err != nil {
+		return true
+	}
+
+	attributes, err := syscall.GetFileAttributes(lpFileName)
+
+	if err != nil {
+		return true
```

Comment:
I'm a bit averse to passing a filter if there's an error while testing it; it may let unintended files through.

Personally, I think that we should just return a fail on the filter and perhaps log something about the error.

Reply:
Good question. But is it better to transfer too much, or too little?  (I.e. to fall back to passing, or to not passing).  E.g. maybe better to transfer too much?  (Especially if, say, after the transfer, the user is about to delete the source!)

I'd be very interested to hear what @zezha-msft thinks, as original architect of the filter interface.  Should there be an error return added to the function signature.

Also, we need to think about the case where the file has been deleted or renamed, between the time the enumeration process saw it, and the time it got filtered.  (I'm not sure how big that time gap may be).  

In any case, once we've thought about all of the above, it's possible the right Boolean to return isn't a hard coded true or false, but rather `resultIfMatch` or `!resultIfMatch` - i.e. the right response to a failure maybe opposite for include vs exclude filters.

Thanks @adreed-msft for seeing this issue.

Reply:
I added a "TODO" comment. 😉

Reply:
That's a great question. I think the current behavior is fine: if the file was deleted (or changed somehow so that getting attributes fails), this filter would not be the right place to throw that error and say the transfer should not proceed.

From the design perspective, if a filter fails to perform its check, it should act like it didn't exist in the first place. In general, the strategy is to be as fault tolerant as possible, and avoid unnecessary support escalations.